core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -951,6 +951,43 @@ wasi_path_remove_directory(wasm_exec_env_t exec_env, wasi_fd_t fd,
     return wasmtime_ssp_path_remove_directory(curfds, fd, path, path_len);
 }
 
+static __wasi_timestamp_t
+get_timeout_for_poll_oneoff(const wasi_subscription_t *in,
+                            uint32 nsubscriptions)
+{
+    __wasi_timestamp_t timeout = 0;
+    const __wasi_subscription_t *clock_subscription = NULL;
+
+    for (size_t i = 0; i < nsubscriptions; ++i) {
+        const __wasi_subscription_t *s = &in[i];
+        if (s->u.type == __WASI_EVENTTYPE_CLOCK
+            && (s->u.u.clock.flags & __WASI_SUBSCRIPTION_CLOCK_ABSTIME) == 0) {
+            clock_subscription = s;
+            break;
+        }
+    }
+    if (clock_subscription != NULL) {
+        __wasi_timestamp_t ts = clock_subscription->u.u.clock.timeout;
+        timeout = ts >= UINT64_MAX ? (__wasi_timestamp_t)-1 : ts;
```

Comment:
Seems we could change it to `timeout = ts` directly?
__wasi_timestamp_t is defined as uint64_t:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/include/wasmtime_ssp.h#L302
 when `ts >= UINT64_MAX`, ts must be `UINT64_MAX`, same as `(__wasi_timestamp_t)-1`.